packages/react-router-native/index.tsx

```diff
@@ -158,35 +160,38 @@ const URLEventType = 'url';
 /**
  * Enables support for the hardware back button on Android.
  */
-export function useHardwareBackButton() {
+export function useAndroidBackButton() {
+  const index = useIndex();
+  const navigate = useNavigate();
+
+  const handleHardwardBackPress = React.useCallback(() => {
+    if (index === 0) {
+      return false; // home screen
+    } else {
+      navigate(-1);
+      return true;
+    }
+  }, [index, navigate]);
+
+  const handleHardwardBackPressRef = React.useRef(handleHardwardBackPress);
```

Comment:
I think you should be ok to skip using this ref and instead just pass `handleHardwareBackPress` directly to the second `useEffect` in the dependency array. Would you agree?

Reply:
My thought process was the following: `BackHandler` allows multiple listeners to be added and then calls them in order from newest to oldest (until one returns `true`). If this listener is added and removed on every `index` change, the order of `BackHandler` listeners might change – and for me as a user that is something I would not expect to happen. So I opted for a solution where the listener is only added on mount and removed on unmount – and that is why we need the `ref` here (or is there another way to achieve the same behavior?). Does that sound good or would you prefer to add/remove the listener on every change?

(Removing the `ref` also breaks the test, since it is counting how often the listener is added and removed. 😃 If you agree with the implementation I will add a comment why the `ref` is needed.)